Emby.Server.Implementations/Library/MediaStreamSelector.cs

```diff
@@ -57,14 +57,18 @@ public static class MediaStreamSelector
             {
                 // Prefer embedded metadata over smart logic
 
-                stream = streams.FirstOrDefault(s => s.IsForced && string.Equals(s.Language, audioTrackLanguage, StringComparison.OrdinalIgnoreCase)) ??
-                    streams.FirstOrDefault(s => s.IsForced) ??
-                    streams.FirstOrDefault(s => s.IsDefault);
+                stream = streams.FirstOrDefault(s => s.IsExternal && s.IsForced && string.Equals(s.Language, audioTrackLanguage, StringComparison.OrdinalIgnoreCase)) ??
+                         streams.FirstOrDefault(s => s.IsExternal && s.IsForced) ??
+                         streams.FirstOrDefault(s => s.IsExternal && s.IsDefault) ??
+                         streams.FirstOrDefault(s => s.IsExternal) ??
+                         streams.FirstOrDefault(s => s.IsForced && string.Equals(s.Language, audioTrackLanguage, StringComparison.OrdinalIgnoreCase)) ??
+                         streams.FirstOrDefault(s => s.IsForced) ??
+                         streams.FirstOrDefault(s => s.IsDefault);
```

Comment:
This is a lot of looping

Reply:
Would it make sense to to an orderby instead?

Reply:
Sure. The advantage of this approach is readability; it's perfectly clear what is prioritized with the code like this. Using a one-time pre-sorted collection will be faster (even though we're talking literal microseconds here), but will be far less readable. I'll have to rewrite most of the subtitle selection code. Is that preferable? If so, I'll get on that.

Reply:
Just ordering by IsExternal and running the old logic could end in a case where non-external subtitles matching IsForced would come before an external subtitle.

You could filter streams into externalStreams and embeddedStreams and run the logic on them sequentially to avoid having so many lines of FirstOrDefault calls, but you don't save any loops and it adds complexity.

Reply:
Thinking about this more, I think it would be cleaner/more maintainable to first filter for external subs, then run the stream selection logic on just the external subs (if there are any) or the full list. That way there isn't duplicated logic, and it's clear that external subs are prioritized. Something like this (untested):
```c#
// only consider external streams if present
var filteredStreams = streams.Any(s => s.IsExternal) ? streams.Where(s => s.IsExternal) : streams;
// treat external streams as default
stream = filteredStreams.FirstOrDefault(s => s.IsForced && string.Equals(s.Language, audioTrackLanguage, StringComparison.OrdinalIgnoreCase)) ??
         filteredStreams.FirstOrDefault(s => s.IsForced) ??
         filteredStreams.FirstOrDefault(s => s.IsDefault || s.IsExternal);
```

The logic works here because an external sub will be chosen if there is one, so it would never fall through to the embedded subs.